src/repos/FlyerRepo.ts

```diff
@@ -0,0 +1,229 @@
+import Filter from 'bad-words';
+import { ObjectId } from 'mongodb';
+import Fuse from 'fuse.js';
+import { Flyer, FlyerModel } from '../entities/Flyer';
+import {
+  DEFAULT_LIMIT,
+  MAX_NUM_DAYS_OF_TRENDING_ARTICLES,
+  FILTERED_WORDS,
+  DEFAULT_OFFSET,
+} from '../common/constants';
+import { OrganizationModel } from '../entities/Organization';
+
+const { IS_FILTER_ACTIVE } = process.env;
+
+function isFlyerFiltered(flyer: Flyer) {
+  if (IS_FILTER_ACTIVE === 'true') {
+    if (flyer.isFiltered) {
+      // If the body has been checked already in microservice
+      return true;
+    }
+    const filter = new Filter({ list: FILTERED_WORDS });
+    return filter.isProfane(flyer.title);
+  }
+  return false;
+}
+
+const getFlyerByID = async (id: string): Promise<Flyer> => {
+  return FlyerModel.findById(new ObjectId(id)).then((flyer) => {
+    if (!isFlyerFiltered(flyer)) {
+      return flyer;
+    }
+    return null;
+  });
+};
+
+const getFlyersByIDs = async (ids: string[]): Promise<Flyer[]> => {
+  return Promise.all(ids.map((id) => FlyerModel.findById(new ObjectId(id)))).then((flyers) => {
+    // Filter out all null values that were returned by ObjectIds not associated
+    // with Flyers in database
+    return flyers.filter((flyer) => flyer !== null && !isFlyerFiltered(flyer));
+  });
+};
+
+const getAllFlyers = async (offset = DEFAULT_OFFSET, limit = DEFAULT_LIMIT): Promise<Flyer[]> => {
+  return FlyerModel.find({})
+    .sort({ date: 'desc' })
+    .skip(offset)
+    .limit(limit)
+    .then((flyers) => {
+      return flyers.filter((flyer) => !isFlyerFiltered(flyer));
```

Comment:
why do you check if it's not null here but in the above and below functions?

Reply:
i think getAllFlyers shouldn't return extra nulls because it's using find({}) and getFlyerByID is only looking for one article so it'll only return null if there aren't any flyers to be found